deploy/helm/edot-collector/kube-stack/values.yaml

```diff
@@ -841,75 +518,414 @@ collectors:
+                - key: agent.name # set via elastictraceprocessor
+              attributes:
+                - key: transaction.root
+                - key: transaction.type
+                - key: metricset.name
+                  default_value: service_transaction
+                - key: elasticsearch.mapping.hints
+                  default_value: [aggregate_metric_double]
+              conditions:
+                - attributes["event.outcome"] != nil and attributes["event.outcome"] == "success"
+              unit: us
+              histogram:
+                buckets: [1]
+                count: Int(AdjustedCount())
+                value: Int(AdjustedCount())
+            - name: event.success_count
+              description: Success count as a metric for service transaction
+              include_resource_attributes:
+                - key: service.name
+                - key: deployment.environment # service.environment
+                - key: telemetry.sdk.language # service.language.name
+                - key: agent.name # set via elastictraceprocessor
+              attributes:
+                - key: transaction.root
+                - key: transaction.type
+                - key: metricset.name
+                  default_value: service_transaction
+                - key: elasticsearch.mapping.hints
+                  default_value: [aggregate_metric_double]
+              conditions:
+                - attributes["event.outcome"] != nil and attributes["event.outcome"] != "success"
+              unit: us
+              histogram:
+                buckets: [0]
+                count: Int(AdjustedCount())
+                value: Double(0)
+      receivers:
+        otlp:
+          protocols:
+            grpc:
+              endpoint: ${env:MY_POD_IP}:4317
+            http:
+              endpoint: ${env:MY_POD_IP}:4318
+      processors:
+        # [Elastic Infra Metrics Processor](https://github.com/elastic/opentelemetry-collector-components/tree/main/processor/elasticinframetricsprocessor)
+        elasticinframetrics:
+          add_system_metrics: true
+          add_k8s_metrics: true
+          drop_original: true
+        # [Attributes Processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/attributesprocessor)
+        attributes/dataset:
+          actions:
+            - key: event.dataset
+              from_attribute: data_stream.dataset
+              action: upsert
+        batch:
+          send_batch_size: 1000
+          timeout: 1s
+          send_batch_max_size: 1500
+        # [Elastic Trace Processor](https://github.com/elastic/opentelemetry-collector-components/tree/main/processor/elastictraceprocessor)
+        elastictrace: {} # The processor enriches traces with elastic specific requirements.
+        # [LSM Interval Processor](https://github.com/elastic/opentelemetry-collector-components/tree/main/processor/lsmintervalprocessor)
+        lsminterval:
```

Comment:
We should a `directory` here, do we have provision for a volume mount?

Reply:
Yes, we can define volumes/volumes mount for each set of collectors. If persistence is not a requirement, we could use an `emptyDir` [volume type](https://kubernetes.io/docs/concepts/storage/volumes/#emptydir)

Does it need to be persistent over pod restarts? And unique per container?

Reply:
Persistence is not necessary for now but we would need to have some headroom to scale the volume. If the node size is big enough then it should be good to start with emptyDir.

Reply:
> If the node size is big enough then it should be good to start with emptyDir.

The issue is that we cannot do any assumption on the node size, we are not currently setting any hardware limit aside from the resource memory limits. The `emptyDir` volume size can be limited (I am a bit lost of which should be the limit), but if we don't set a limit it will be memory backed (max size == pod.resource.memory.limit) which we might not get any benefit out of it.

Reply:
> but if we don't set a limit it will be memory backed (max size == pod.resource.memory.limit) which we might not get any benefit out of it.

Hmm, we definitely don't want memory backed. I think we can start with 20GB of disk space if the node will allow that, does that sound reasonable for now?

Reply:
I'm a bit concerned about adding this hardware requirement at the moment. While 20GB of disk space sounds reasonable in principle, the configuration is deploying two instances of the gateway collector (40GB in total). Also, I think we should add this disk requisite on the onboarding page (Kibana) as well. 
Do you think we could explore an opt-in approach for the next iteration?

Reply:
> Do you think we could explore an opt-in approach for the next iteration?

Hmm, it is already kinda opt-in (users can define the directory, if not defined we use in-memory).

> While 20GB of disk space sounds reasonable in principle, the configuration is deploying two instances of the gateway collector (40GB in total).

Even 40GB sounds reasonable to me since disk is cheap but I might be missing something. Overall, opt-in sounds okay to me for now.